### PR TITLE
[GraFx Developers] - media connectors

### PR DESCRIPTION
**Pull request by seancrowe**
Still in draft, but good

Latest build:  [https://wonderful-sky-012b9bc03-393.westeurope.1.azurestaticapps.net/](https://wonderful-sky-012b9bc03-393.westeurope.1.azurestaticapps.net/)

Do not bookmark the doc link, it will change!

### CODE REVIEW

File: docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+2. Add a new image variable to your document.
+3. In the image selection interface, choose your newly published Connector.
+4. Look for the new setting: "Number (1 to 100) of Images to Display".
+5. Adjust this setting to different values within the allowed range (1 to 100).
+6. Click "Select image" to observe how the number of displayed images changes based on your setting.
+
+## Adding a Rectangle/Square Image Setting
+
+
+The picsum.photos API supports both outputting rectangular and square images. We can enhance our connector by adding a setting that allows Designes to choose between these two image crops. This setting will affect how images are displayed in both the image selector and the document frame.
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md:152]: Spelling, change "Designes" to "Designers"

File: docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      {
+        name: "wide",
+        displayName: "Display images as rectangluar instead of square",
+        type: "boolean"
+      }
+    ];
+  }
+  ```
+
+This addition introduces a new boolean option named "wide" that users can toggle to switch between rectangular and square image displays.
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md:173]: Everything here is still in the code block!

File: docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      }
+      default: {
+        const picture = await this.runtime.fetch(`https://picsum.photos/id/${id}/${(context.wide) ? "2000/" : ""}1000`, { method: "GET" });
+        return picture.arrayBuffer;
+      }
+    }
+  }
+```
+
+Key changes in this method:
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md:201]: Not being read as numbered list, all one paragraph

File: docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+1. Publish your updated connector using the method described in the previously.
+2. In your document, add or select an existing image variable using this Connector.
+3. Set up an image frame with the mode set to "Fit" and linked to the variable.
+4. Locate the new "Display images as rectangular instead of square" setting in the connector options.
+5. Toggle this setting and observe how it affects the images in both the selector and the document frame.
+
+This demonstrates that our connector is not just affecting the image selection process, but also how the image is displayed within the document itself.
+
+### Key Accomplishments
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md:218]: This should all be in the add-environment-options doc?

Also, the next steps link at the bottom of this page goes to building connector with auth (not there yet, already know that), it should have a link to the environment options doc instead as next step?

File: docs/GraFx-Developers/connectors/media-connector/add-environment-options-to-your-connector/index.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    There is a known issue in the Connector CLI tool where default values are not applied correctly.
+
+## Implementing the Option in Code
+
+### Updating the `query` Method
+
+In the `query` method, access the new option via the `runtime` object. Update the fetch request by replacing the hardcoded base URL with the new option:
+
+```typescript
+const resp = await this.runtime.fetch(`https://${this.runtime.options["baseURL"]}/v2/list?page=1&limit=${limit}`, {
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/media-connector/add-environment-options-to-your-connector/index.md:42]: This assumes that you've done the variable options step, though that's not explicitly in the prereqs for this (this has the limit param in the URL) and it's also not set up in the provided git repo.

Either the limit param should be removed from the fetch URL here, or the prereqs should include the variable options section and have an updated git repo included to start from

File: docs/GraFx-Developers/connectors/media-connector/add-environment-options-to-your-connector/index.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+```typescript
+async download(
+  id: string,
+  previewType: Media.DownloadType,
+  intent: Media.DownloadIntent,
+  context: Connector.Dictionary
+): Promise<Connector.ArrayBufferPointer> {
+  switch (previewType) {
+    case "thumbnail": {
+      const picture = await this.runtime.fetch(`https://${this.runtime.options["baseURL"]}/id/${id}/${context.wide ? "400/" : ""}200`, { method: "GET" });
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/media-connector/add-environment-options-to-your-connector/index.md:60]: same as above, this has the option for wide images which isn't set up in the base repo

File: docs/GraFx-Developers/connectors/authorization-for-connectors/index.md [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+## Authorization Limitations
+
+While the Connector framework supports the most common authorization schemas, it's important to note that completely customizable authorization is not possible. If your specific authorization requirements don't align with the four provided types, you may need to develop a workaround solution.
+
+For example, services like GraFx Media and GraFx Platform don't meet the requirements of any of the supported authorization types. In such cases, you would need to implement an intermediary service that adapts the authorization workflow to work within the Connector framework.
+
+## Define Supported Authorization Types
+Before implementing authorization, you must define the supported authorization types in your `package.json` file. This ensures consistency between your CLI commands and your package configuration.
+
+To do so, add or modify the `supportedAuth` property you `package.json` file with the types of authentication your Connector supports.
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/authorization-for-connectors/index.md:27]: Grammar, change to "To do so, add or modify the `supportedAuth` property in your `package.json` file with the types of authentication your Connector supports."

File: docs/GraFx-Developers/connectors/authorization-for-connectors/index.md [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+```
+
+!!! tip "Matching Environment"
+
+    Ensure that the environment in your base URL matches the environment passed via the `-e` argument.
+
+
+### Authorization Usage (`-au`)
+
+This argument expects one of two value that will specify the usage scenario for the authorization.
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/authorization-for-connectors/index.md:94]: Grammar, change to "This argument expects one of two values that will specify the usage scenario for the authorization."

File: docs/GraFx-Developers/connectors/authorization-for-connectors/index.md [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+
+Expected values:
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/authorization-for-connectors/index.md:96]: This isn't reading as a list, all just one paragraph right now

File: docs/GraFx-Developers/connectors/authorization-for-connectors/index.md [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+2. The `grant_type` for OAuth types is automatically set as per the specification and cannot be configured.
+3. `access_token` and `refresh_token` are cached internally for OAuth types. There's currently no way to reset them without deleting the Connector.
+4. OAuth Authorization Code is user-specific and not useful with an Integration User (used outside the GraFx Platform).
+5. OAuth Authorization Code is limited to "browser" usage, which may restrict its applicability in certain scenarios.
+
+### Authorization Data File Requirements (`--auth-data-file`)
+
+Each authorization type requires a specific JSON schema which is related to the requirements of that authorization type. Keep in mind that the `grant_type` for OAuth types is predefined by the specifications.
+
+Each authorization type requires a specific JSON schema. The `grant_type` for OAuth types is predefined by the specifications.
```
Comment (bhead43) [inline docs/GraFx-Developers/connectors/authorization-for-connectors/index.md:141]: This is a copy of the previous paragraph, just a teensy bit less detailed. Choose one to stay and one to shuffle off this mortal coil

File: docs/GraFx-Developers/connectors/media-connector/build-a-simple-media-connector/index.md [hunk 11]
```diff
@@ -0,0 +1,409 @@
...
+  }
+
+  // Handle error case
+  throw new Error("Failed to fetch images from picsum.photos");
+}
+```
+
+## Implementing the Detail Method
+
+The `detail` method is utilized for defining the initial width and height of the image frame when selected from the Media panel. While this method requires several parameters, only the width and height are actually used for this purpose. Even though, width and height are not required.
```
[the remark this answers is not in the document]
Comment (seancrowe) [reply docs/GraFx-Developers/connectors/media-connector/build-a-simple-media-connector/index.md:338]: It's just confusing in general 😜

File: docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md (hunk 4 above)
Comment (bhead43) [reply docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md:219]: This section needs to be moved to environment options

**Review by bhead43 — APPROVED** [review]

**Pull request merged**